### PR TITLE
Bugfixes for double spend spams. 

### PR DESCRIPTION
**Pull request by piotrm50**
This PR introduces a bunch of updates found when running double spend spams in local Docker network.

### CODE REVIEW

**Review by muXxer — CHANGES_REQUESTED** [review]

File: pkg/protocol/engine/tipselection/v1/provider.go [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
 // NewProvider creates a new TipSelection provider, that can be used to inject the component into an engine.
 func NewProvider(opts ...options.Option[TipSelection]) module.Provider[*engine.Engine, tipselection.TipSelection] {
 	return module.Provide(func(e *engine.Engine) tipselection.TipSelection {
-		t := New(e.TipManager, e.Ledger.ConflictDAG(), e.Ledger.MemPool(), e.EvictionState.LatestRootBlocks, opts...)
+		t := New(e, e.TipManager, e.Ledger.ConflictDAG(), e.Ledger.MemPool(), e.EvictionState.LatestRootBlocks, opts...)
 
 		e.HookConstructed(func() {
 			e.Ledger.HookInitialized(func() {
+				e.Events.AcceptedBlockProcessed.Hook(func(block *blocks.Block) {
+					livenessThresholdDuration := time.Duration(uint64(e.LatestAPI().ProtocolParameters().LivenessThreshold())*uint64(e.LatestAPI().ProtocolParameters().TimeProvider().SlotDurationSeconds())) * time.Second
```
[the remark this answers is not in the document]
Comment (piotrm50) [reply pkg/protocol/engine/tipselection/v1/provider.go:21]: Is it ok if I just move it 2 lines above, outside of the AcceptedBlockProcessed handler?

[the remark this answers is not in the document]
Comment (alexsporn) [reply pkg/protocol/engine/tipselection/v1/provider.go:21]: `LatestAPI()` is wrong btw, you should use `CurrentAPI()`
Latest could be an already planned protocol upgrade

[the remark this answers is not in the document]
Comment (piotrm50) [reply pkg/protocol/engine/tipselection/v1/provider.go:21]: > LatestAPI() is wrong btw, you should use CurrentAPI()
Changed that.

File: components/debugapi/params.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package debugapi
+
+import (
+	"github.com/iotaledger/hive.go/app"
+)
+
+// ParametersDebugAPI contains the definition of configuration parameters used by the debug API.
+type ParametersDebugAPI struct {
+	// Enabled whether the DebugAPI component is enabled.
+	Enabled bool `default:"true" usage:"whether the BlockIssuer component is enabled"`
```
Comment (muXxer) [inline components/debugapi/params.go:10]: Wrong comment

File: components/debugapi/component.go [hunk 3]
```diff
@@ -73,19 +65,39 @@ type dependencies struct {
...
+		Directory: ParamsDebugAPI.Path,
+
+		Version:      1,
+		PrefixHealth: []byte{0},
+	}, func(err error) {
+		fmt.Printf(">> DebugAPI Error: %s\n", err)
+	}, prunable.WithGranularity(ParamsDebugAPI.DBGranularity), prunable.WithMaxOpenDBs(ParamsDebugAPI.MaxOpenDBs),
+	)
+
 	routeGroup := deps.RestRouteManager.AddRoute("debug/v3")
```
Comment (muXxer) [inline components/debugapi/component.go:85]: Oh, we really have v3 of the debug API already? :D

**Review by muXxer — CHANGES_REQUESTED** [review]

File: components/debugapi/component.go (hunk 3 above)
Comment (muXxer) [reply components/debugapi/component.go]: sorry, just had a look in hornet. there we are at v1 so this here should be v2

**Review by muXxer — APPROVED** [review]

**Pull request merged**